pointcloud/basic_octree_test.go

```diff
@@ -1,33 +1,29 @@
-package octree
+package pointcloud
 
 import (
-	"context"
 	"math"
 	"os"
 	"testing"
 
-	"github.com/edaniels/golog"
 	"github.com/golang/geo/r3"
 	"github.com/pkg/errors"
 	"go.viam.com/test"
 	"go.viam.com/utils/artifact"
-
-	pc "go.viam.com/rdk/pointcloud"
 )
 
 // Helper function for generating a new empty octree.
-func createNewOctree(ctx context.Context, center r3.Vector, side float64, logger golog.Logger) (*basicOctree, error) {
-	octree, err := New(ctx, center, side, logger)
+func createNewOctree(center r3.Vector, side float64) (*BasicOctree, error) {
+	octree, err := NewBasicOctree(center, side)
 	if err != nil {
 		return nil, err
 	}
 
-	basicOct := octree.(*basicOctree)
+	basicOct := octree.(*BasicOctree)
```

Comment:
Having all the `*Octree` functions return types of `BasicOctree`, would prevent you from needing to do this type assertion & I don't think would prevent you from treating the octree as a `PointCloud`, correct me if I'm wrong.

Reply:
Originally it was useful to have an basicOctree be initially returned as a Octree as that interface was linked to the PointCloud interface. Now that we have moved everything into pointcloud, we can cast it specifically to a basicOctree upon creation.